src/converter/jfr2pprof.java

```diff
@@ -116,26 +133,45 @@ public void dump(final OutputStream out) throws Exception {
 
         final Proto sampleType = new Proto(100);
 
-        profile.field(PROFILE_STRING_TABLE, "cpu".getBytes(StandardCharsets.UTF_8));
-        sampleType.field(VALUETYPE_TYPE, stringId++);
+        if (eventClass == AllocationSample.class) {
+            profile.field(PROFILE_STRING_TABLE, "alloc_space".getBytes(StandardCharsets.UTF_8));
+            sampleType.field(VALUETYPE_TYPE, stringId++);
+            profile.field(PROFILE_STRING_TABLE, "nanoseconds".getBytes(StandardCharsets.UTF_8));
+            sampleType.field(VALUETYPE_UNIT, stringId++);
+        } else {
+            profile.field(PROFILE_STRING_TABLE, "cpu".getBytes(StandardCharsets.UTF_8));
+            sampleType.field(VALUETYPE_TYPE, stringId++);
+            profile.field(PROFILE_STRING_TABLE, "nanoseconds".getBytes(StandardCharsets.UTF_8));
+            sampleType.field(VALUETYPE_UNIT, stringId++);
+        }
 
-        profile.field(PROFILE_STRING_TABLE, "nanoseconds".getBytes(StandardCharsets.UTF_8));
-        sampleType.field(VALUETYPE_UNIT, stringId++);
+        profile.field(PROFILE_STRING_TABLE, "allocation_size".getBytes(StandardCharsets.UTF_8));
+        final int allocationSizeKeyId = stringId++;
+        profile.field(PROFILE_STRING_TABLE, "bytes".getBytes(StandardCharsets.UTF_8));
+        final int allocationSizeUnitId = stringId++;
 
         profile.field(PROFILE_SAMPLE_TYPE, sampleType);
 
-        final List<ExecutionSample> jfrSamples = reader.readAllEvents(ExecutionSample.class);
+        final List<? extends Event> jfrSamples = reader.readAllEvents(eventClass);
         final Dictionary<StackTrace> stackTraces = reader.stackTraces;
         long previousTime = reader.startTicks; // Mutate this to keep track of time deltas
 
+        final Proto allocationLabel = new Proto(16)
+                .field(LABEL_KEY, allocationSizeKeyId)
+                .field(LABEL_NUM_UNIT, allocationSizeUnitId);
+
         // Iterate over samples
-        for (final ExecutionSample jfrSample : jfrSamples) {
+        for (final Event jfrSample : jfrSamples) {
             final StackTrace stackTrace = stackTraces.get(jfrSample.stackTraceId);
             final long[] methods = stackTrace.methods;
             final byte[] types = stackTrace.types;
 
+            final Proto sample = new Proto(1_000);
             final long nanosSinceLastSample = (jfrSample.time - previousTime) * 1_000_000_000 / reader.ticksPerSec;
-            final Proto sample = new Proto(1_000).field(SAMPLE_VALUE, nanosSinceLastSample);
+            sample.field(SAMPLE_VALUE, nanosSinceLastSample);
+            if (eventClass == AllocationSample.class) {
+                sample.field(SAMPLE_LABEL, allocationLabel.field(LABEL_NUM, jfrSample.value()));
```

Comment:
`field` method mutates the object; it's not correct to reuse `allocationLabel` this way.